cmd/uzfs_test/uzfs_rebuilding.c

```diff
@@ -0,0 +1,565 @@
+	make_vdev(file_name);
+
+	err = uzfs_create_pool(pool_name, file_name, &t_spa);
+	if (t_spa == NULL || err != 0) {
+		printf("shouldn't create pool(%s) with non existing"
+		    " disk..err(%d)\n", pool_name, err);
+		exit(1);
+	}
+
+	err = uzfs_create_dataset(t_spa, zvol_name, vol_size,
+	    block_size, &t_zvol);
+	if (t_zvol == NULL || err != 0) {
+		printf("creating ds(%s) errored %d..\n", zvol_name, err);
+		exit(1);
+	}
+
+	*spa = t_spa;
+	*zvol = t_zvol;
+
+	strfree(pool_name);
+	strfree(file_name);
+	strfree(zvol_name);
+}
+
+void
+uzfs_rebuild_test(void *arg)
+{
+	uzfs_test_info_t *test_info = (uzfs_test_info_t *)arg;
+	kmutex_t mtx;
+	kcondvar_t cv;
+	int threads_done = 0, num_threads = 0;
+	kthread_t *writer, *reader;
+	worker_args_t writer_args, **reader_args;
+	int reader_count;
+	int n = 0;
+
+	printf("starting %s\n", test_info->name);
+
+	while (n++ < test_iterations) {
+		create_and_init_pool(&spa1, &zvol1);
+		create_and_init_pool(&spa2, &zvol2);
+
+		mutex_init(&mtx, NULL, MUTEX_DEFAULT, NULL);
+		cv_init(&cv, NULL, CV_DEFAULT, NULL);
+
+		writer_args.zv = NULL;
+		writer_args.threads_done = &threads_done;
+		writer_args.mtx = &mtx;
+		writer_args.cv = &cv;
+		writer_args.io_block_size = io_block_size;
+		writer_args.active_size = active_size;
+
+		writer = zk_thread_create(NULL, 0,
+		    (thread_func_t)replica_writer_thread, &writer_args, 0, NULL,
+		    TS_RUN, 0, PTHREAD_CREATE_DETACHED);
+		num_threads++;
+
+		mutex_enter(&mtx);
+		while (threads_done != num_threads)
+			cv_wait(&cv, &mtx);
+		mutex_exit(&mtx);
+
+		reader_count = 4;
+		int i = 0;
+
+		reader_args = umem_alloc(sizeof (*reader_args) * reader_count,
+		    UMEM_NOFAIL);
+
+		for (i = 0; i < reader_count; i++) {
+			worker_args_t *r_arg;
+			struct replica_read_data *r_data;
+
+			r_arg = umem_alloc(sizeof (*r_arg), UMEM_NOFAIL);
+			r_data = umem_alloc(sizeof (*r_data), UMEM_NOFAIL);
+
+			r_data->offset = i * ((vol_size) / reader_count);
+			r_data->len = (vol_size) / reader_count;
+			r_arg->zv = r_data;
+			r_arg->threads_done = &threads_done;
+			r_arg->mtx = &mtx;
+			r_arg->cv = &cv;
+			r_arg->io_block_size = io_block_size;
+			r_arg->active_size = active_size;
+			reader_args[i] = r_arg;
+
+			reader = zk_thread_create(NULL, 0,
+			    (thread_func_t)replica_reader_thread,
+			    reader_args[i], 0, NULL, TS_RUN, 0,
+			    PTHREAD_CREATE_DETACHED);
+			num_threads++;
+		}
+
+		mutex_enter(&mtx);
+		while (threads_done != num_threads)
+			cv_wait(&cv, &mtx);
+		mutex_exit(&mtx);
+
+		for (i = 0; i < reader_count; i++) {
+			worker_args_t *r_arg;
+
+			r_arg  = reader_args[i];
+			umem_free(r_arg->zv, sizeof (*r_arg->zv));
+			umem_free(r_arg, sizeof (*r_arg));
+		}
+
+		umem_free(reader_args, sizeof (*reader_args) * reader_count);
+
+		cv_destroy(&cv);
+		mutex_destroy(&mtx);
+
+		uzfs_close_dataset(zvol1);
+		uzfs_close_dataset(zvol2);
+		uzfs_close_pool(spa1);
+		uzfs_close_pool(spa2);
+		printf("%s pass:%d\n", test_info->name, n);
+	}
```

Comment:
@jkryl  it is already there for read case. In replica_reader_thread if there is any mismatch in data between two replica then it will print errors and exit with 1.
For write purpose, i will add.

Reply:
thx. Write and verify cases are missing this as of now.

Reply:
for verify, we don't need. VERIFY will abort code in false cases. right?